doc/additional-configuration.md

```diff
@@ -0,0 +1,123 @@
+# Additional configuration for DevWorkspaces
+
+DevWorkspaces allow for additional configuration through the use Devfile attributes and Kubernetes labels and annotations. There are three fields in the DevWorkspace that are used to configure functionality:
+* Devfile attributes defined within the DevWorkspace spec, either at the *top-level* (`.spec.template.attributes`) or on *individual components*. For example:
+    ```yaml
+    kind: DevWorkspace
+    apiVersion: workspace.devfile.io/v1alpha2
+    metadata:
+      name: my-workspace
+    spec:
+      template:
+        attributes:
+          my-attribute: my-attribute-value
+    ```
+* Kubernetes metadata labels (`.metadata.labels`):
+    ```yaml
+        kind: DevWorkspace
+        apiVersion: workspace.devfile.io/v1alpha2
+        metadata:
+          name: my-workspace
+          labels:
+            my-label: my-label-value
+    ```
+* Kubernetes metadata annotations (`.metadata.annotations`):
+    ```yaml
+        kind: DevWorkspace
+        apiVersion: workspace.devfile.io/v1alpha2
+        metadata:
+          name: my-workspace
+          labels:
+            my-annotation: my-annotation-value
+    ```
+
+## Restricting access to DevWorkspaces
+Applying the
+```yaml
+controller.devfile.io/restricted-access: "true"
+```
+annotation to a DevWorkspace enables additional access control for the workspace. When this annotation is applied to a DevWorkspace:
+* Only the user that created the DevWorkspace can access a terminal in the workspace via `pods/exec`
+* Only the DevWorkspace Operator serviceaccount or the user that created the DevWorkspace can modify fields in the DevWorkspace custom resource.
+
+This is useful in case a DevWorkspace is expected to contain sensitive information.
+
+
+## Configuring persistent storage used for a DevWorkspace
+The top-level Devfile attribute `controller.devfile.io/storage-type` can be used to configur persistent storage for DevWorkspaces:
+```yaml
+kind: DevWorkspace
+apiVersion: workspace.devfile.io/v1alpha2
+metadata:
+  name: my-workspace
+spec:
+  template:
+    attributes:
+      controller.devfile.io/storage-type: <storage-type>
+```
+where `<storage-type>` is one of
+- `common`: Use one PVC for all workspace volumes, mounting Devfile volumes in subpaths within the common PVC
+- `ephemeral`: Replace all volumes with `emptyDir` volumes. This storage type is non-persistent; any local changes will be lost when the workspace is stopped. This is the equivalent of marking all volumes in the Devfile as `ephemeral: true`
+- `async`: Use emptyDir volumes for workspace volumes, but include a sidecar that synchronises local changes to a persistent volume as in the `common` strategy. This can potentially avoid issues where mounting volumes to a workspace on start takes a long time.
```

Comment:
```suggestion
- `async`: Use `emptyDir` volumes for workspace volumes, but include a sidecar that synchronises local changes to a persistent volume as in the `common` strategy. This can potentially avoid issues where mounting volumes to a workspace on startup takes a long time.
```